test/integration/http_integration.cc

```diff
@@ -205,8 +206,23 @@ HttpIntegrationTest::makeHttpConnection(Network::ClientConnectionPtr&& conn) {
 HttpIntegrationTest::HttpIntegrationTest(Http::CodecClient::Type downstream_protocol,
                                          Network::Address::IpVersion version,
                                          const std::string& config)
-    : BaseIntegrationTest(version, config), downstream_protocol_(downstream_protocol) {
-  // Legacy integration tests expect the default listener to be named "http" for lookupPort calls.
+    : HttpIntegrationTest::HttpIntegrationTest(downstream_protocol,
+                                               /*upstream_address_fn=*/
+                                               [version](int) {
+                                                 return Network::Utility::parseInternetAddress(
+                                                     Network::Test::getAnyAddressString(version),
+                                                     0);
+                                               },
+                                               version, config) {}
+
+HttpIntegrationTest::HttpIntegrationTest(
+    Http::CodecClient::Type downstream_protocol,
+    std::function<Network::Address::InstanceConstSharedPtr(int)> upstream_address_fn,
```

Comment:
can you factor out this type using a 'using' statement in BaseIntegrationTest's scope?

Reply:
Done.